owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsPasscode.kt

```diff
@@ -0,0 +1,226 @@
+
+import android.app.Activity
+import android.content.Intent
+import android.preference.PreferenceManager
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Espresso.pressBack
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.ActivityResultMatchers.hasResultCode
+import androidx.test.espresso.contrib.ActivityResultMatchers.hasResultData
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasExtra
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import com.owncloud.android.R
+import com.owncloud.android.ui.activity.PassCodeActivity
+import org.junit.After
+import org.junit.Assert.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class OCSettingsPasscode {
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(PassCodeActivity::class.java, true, false)
+
+    val intent = Intent()
+    val errorMessage = "PassCode Activity error"
+    private val KEY_PASSCODE = "KEY_PASSCODE"
+    private val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    @After
+    fun tearDown() {
+        //Clean preferences
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear().commit()
+    }
+
+
+    @Test
+    fun passcodeView(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withId(R.id.header)).check(matches(isDisplayed()))
+        onView(withId(R.id.explanation)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code_explanation)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt0)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt1)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt2)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt3)).check(matches(isDisplayed()))
+        onView(withId(R.id.cancel)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun firstTry(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
```

Comment:
Save `arrayOf('1','1','1','1')` in a constant at the top of the class

Reply:
Done 👍